alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/binder/solr/tracking/SolrTrackingMetrics.java

```diff
@@ -0,0 +1,54 @@
+package eu.xenit.alfred.telemetry.binder.solr.tracking;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import java.util.function.ToDoubleFunction;
+import org.alfresco.repo.solr.SOLRTrackingComponent;
+import org.alfresco.repo.transaction.RetryingTransactionHelper;
+import org.alfresco.service.transaction.TransactionService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SolrTrackingMetrics {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SolrTrackingMetrics.class);
+
+    public static final String SOLR_METRICS_PREFIX = "solr.tracking";
+    private TransactionService transactionService;
+
+    private SOLRTrackingComponent solrTrackingComponent;
+    private MeterRegistry registry;
+
+    public SolrTrackingMetrics(SOLRTrackingComponent solrTrackingComponent, TransactionService transactionService,
+            MeterRegistry registry, boolean enabled) {
+        this.solrTrackingComponent = solrTrackingComponent;
+        this.transactionService = transactionService;
+        this.registry = registry;
+        if (enabled) {
+            registerMetrics();
+        }
+    }
+
+    private void registerMetrics() {
+        LOGGER.info("Registering Solr metrics");
+        registerSolrTrackingMetric("maxTxnId",
+                solrTrackingComponent1 -> solrTrackingComponent1.getMaxTxnId());
```

Comment:
Possible to replace with method reference? (Would make it a bit cleaner?)

Reply:
yes